src/ai/backend/storage/api/manager.py

```diff
@@ -776,14 +777,18 @@ class Params(TypedDict):
     ) as params:
         await log_manager_api_entry(log, "mkdir", params)
         ctx: RootContext = request.app["ctx"]
+        relpaths = params["relpaths"]
+        vfid = params["vfid"]
+        parents = params["parents"]
+        exist_ok = params["exist_ok"]
+
         async with ctx.get_volume(params["volume"]) as volume:
-            with handle_fs_errors(volume, params["vfid"]):
-                await volume.mkdir(
-                    params["vfid"],
-                    params["relpath"],
-                    parents=params["parents"],
-                    exist_ok=params["exist_ok"],
-                )
+            with handle_fs_errors(volume, vfid):
+                mkdir_tasks = [
+                    volume.mkdir(vfid, relpath, parents=parents, exist_ok=exist_ok)
+                    for relpath in relpaths
+                ]
+                await asyncio.gather(*mkdir_tasks)
```

Comment:
It is a good idea to use `asyncio.gather()` here.
How about passing `True` to [`return_exceptions`](https://docs.python.org/3/library/asyncio-task.html#asyncio.gather) parameter and handle errors here?

Reply:
The reason why I suggest to use `return_exception` is that the current implement returns only one failure when multiple `mkdir` fails.